api/task_get_test.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/api/oapigen"
+	"github.com/hashicorp/consul-terraform-sync/config"
+	mocks "github.com/hashicorp/consul-terraform-sync/mocks/server"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTaskLifeCycleHandler_GetTaskByName(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name          string
+		mock          func(*mocks.Server)
```

Comment:
nit, this naming can be improved to indicate this is a server generator/function, specially since we're importing the package `mock` in this file